docker/main/ngen/Dockerfile

```diff
@@ -575,7 +567,16 @@ RUN cd ${WORKDIR}/ngen \
                 $BUILD_DIR/test/test_bmi_multi; \
             fi \
     done \
-    && find cmake_build* -type f -name "*" ! \( -name "*.so" -o -name "ngen" -o -name "partitionGenerator" \) -exec rm {} + 
+    && find cmake_build* -type f -name "*" ! \( -name "*.so" -o -name "ngen" -o -name "partitionGenerator" \) -exec rm {} +
```

Comment:
Hmm....so the intent is to build in all the test binaries in the previous layer and then just use them in the `testing` layer?  This will add some bloat to the ngen image size that may not be wanted long term.

Reply:
I'm not entirely sure why this line registers in the diff.  Maybe I am missing it.  But regarding your question @hellkite500, none of the other layers should be built differently.  I'll be looking into those types of modifications, but that'll be handled separately.

This test layer is only used to build `ngen-build-test` image, defined as a separate service in _docker-build.yml_.  For the main image, this layer is ignored, because no other layer uses it, and it isn't the last layer.

The intent is just to move where in the build process this layer gets built.  That way, it's easier to control how the testing layer is used with minimal re-building of things that haven't really changed.